R/lazy-join-query.R

```diff
@@ -199,15 +199,26 @@ sql_build.lazy_multi_join_query <- function(op, con, ...) {
 }
 
 generate_join_table_names <- function(table_names) {
+  table_name_length_max <- dplyr::coalesce(max(nchar(table_names$name)), 0)
```

Comment:
Why would `max(nchar(table_names$name))` be NA?

Reply:
Currently, there can be cases where `NA` values are found in `table_names$name`. One such example (see below) is when joining an already-joined lazy query with a net new table.

I'm not familiar enough with the SQL composition to know whether each value of `name` _should_ be non-`NA`, but I suspect the current behavior is expected. @mgirlich any thoughts?

Example: after setting `table_name_length_max <- max(nchar(table_names$name))`:
``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(dbplyr)
#> 
#> Attaching package: 'dbplyr'
#> The following objects are masked from 'package:dplyr':
#> 
#>     ident, sql
lf <- dbplyr::lazy_frame(x = 1, a = 1, .name = "df1")
lf2 <- dbplyr::lazy_frame(x = 1, b = 2, .name = "df2")
lf3 <- dbplyr::lazy_frame(x = 1, b = 2, .name = "df3")

df_join_1 <- left_join(lf, lf2, by = "x")

right_join(df_join_1, lf3, by = "x")
#> Error in if (minlength <= 0L) {: missing value where TRUE/FALSE needed
```

<sup>Created on 2023-02-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Stepping inside `generate_join_table_names()`:

```
Browse[1]> table_names
# A tibble: 2 × 2
  name  from  
  <chr> <chr> 
1 NA    ""    
2 df3   "name"
```

Reply:
In the current implementation this is expected but it's probably better to get rid of `NA` in a separate PR.